File: solutions/javascript/secret-handshake/1/secret-handshake.js
const handshakeCommands = Object.freeze([
  'wink',
  'double blink',
  'close your eyes',
  'jump'
]);

export const commands = (handshakeCode) => {
  if (typeof handshakeCode !== 'number')
    throw new Error('Handshake must be a number');

  const handshakes = handshakeCommands.filter(
    (_, index) => handshakeCode & Math.pow(2, index),
  );

  if (handshakeCode & Math.pow(2, 4)) handshakes.reverse();

  return handshakes;
};

File: solutions/go/secret-handshake/1/secret_handshake.go
package secret

const testVersion = 2

var values = []string{"wink", "double blink", "close your eyes", "jump"}

func Handshake(code uint) []string {
	events := make([]string, 0)
	for index := 0; index < len(values); index++ {
		if code&uint(1<<uint(index)) > 0 {
			events = append(events, values[index])
		}
	}
	if code&16 >= 1 {
		for index := len(events)/2 - 1; index >= 0; index-- {
			oppositeIndex := len(events) - 1 - index
			events[index], events[oppositeIndex] = events[oppositeIndex], events[index]
		}
	}
	return events
}
